src/ring_theory/localization.lean

```diff
@@ -1153,6 +1153,152 @@ def at_one [is_localization.away (1 : R) S] : R ≃ₐ[R] S :=
 
 end at_units
 
+section localization_localization
+
+variable (M)
+
+variables (N : submonoid S) (T : Type*) [comm_ring T] [algebra S T]
+variables [algebra R T] [is_scalar_tower R S T]
+
+/--
+Localizing wrt `M ⊆ R` and then wrt `N ⊆ S = M⁻¹R` is equal to the localization of `R` wrt this
+module. See `localization_localization_is_localization`.
+-/
+-- This should only be defined when `S` is the localization `M⁻¹R`, hence the nolint.
+@[nolint unused_arguments]
+def localization_localization_submodule : submonoid R :=
+(N ⊔ M.map (algebra_map R S)).comap (algebra_map R S)
+
+@[simp]
+lemma mem_localization_localization_submodule (x : R) :
```

Comment:
Since the variables appear on either side of the `↔`, we can make them implicit. This will allow a bit of golfing in `map_units`:
```suggestion
variables {M N}
@[simp]
lemma mem_localization_localization_submodule {x : R} :
```